src/App.jsx

```diff
@@ -44,25 +39,35 @@ const App = () => {
 	useEffect(() => {
 		if (isTokenExist) {
 			setIsLoggedIn(true);
-			axios
-				.get('http://localhost:3000/users/me', {
-					headers: {
-						Authorization: localStorage.getItem('userToken'),
-					},
-				})
-				.then((response) => {
-					setUserName(response.data.result.name);
-				});
+			api.getUser().then((data) => {
+				dispatch(
+					loginUser({
+						name: data.result.name,
+						email: data.result.email,
+						token: localStorage.getItem('userToken'),
+					})
+				);
+			});
 		}
-	}, [isLoggedIn]);
+
+		dispatch(setAuthors(fetchedAuthors.data));
+		dispatch(setCourses(fetchedCourses.data));
+	}, [isLoggedIn, fetchedAuthors, fetchedCourses]);
+
+	useEffect(() => {
+		if (fetchedCourses.loading || fetchedAuthors.loading) {
+			setIsLoading(true);
+		}
+
+		let loadingTimer = setTimeout(() => setIsLoading(false), 1000);
+		return () => {
+			clearTimeout(loadingTimer);
+		};
+	}, [isLoading]);
```

Comment:
And this
Avoid timers
```suggestion
	useEffect(() => {
		setIsLoading(coursesLoaded && authorsLoaded);
	}, [coursesLoaded, authorsLoaded]);
```

Reply:
fixed

Reply:
fixed